pd/src/genesis.rs

```diff
@@ -47,15 +47,21 @@ impl Allocation {
     }
 }
 
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct ValidatorPower {
+    pub validator: Validator,
+    pub power: tendermint::vote::Power,
+}
+
 /// The application state at genesis.
 #[derive(Deserialize, Serialize, Debug, Clone)]
 pub struct AppState {
-    /// The initial token allocations.
-    pub allocations: Vec<Allocation>,
     /// The number of blocks in each epoch.
     pub epoch_duration: u64,
     /// The initial validator set.
-    pub validators: Vec<Validator>,
+    pub validators: Vec<ValidatorPower>,
+    /// The initial token allocations.
+    pub allocations: Vec<Allocation>,
```

Comment:
Moving `allocations` to the end of the structure places it last in the JSON output, which is a bit easier to work with (the allocations might be much longer than other genesis data).

Maybe we should define the genesis structures also as proto files?

Reply:
Along the lines of your commentary above, it seems good to move away from ad-hoc serde derivations, so that makes sense to me.